Changed_copyfileslab4/copyfileslab4/Program.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+
+namespace copyfileslab4
+{
+    internal class Program
+    {
+        static void Main(string[] arguments)
+        {
+            if (arguments.Length < 3)
+            {
+                Console.WriteLine("You wrote less than 3 arguments");
+                Console.WriteLine("An example of the correct launch of the program: copyfileslab4.exe \"D:\\Documents\" \"D:\\Games\" \"*.txt\" \"*.pptx\"");
+                Console.WriteLine("Program execution did not complete successfully");
+                return;
+            }
+
+            string Folder1 = arguments[0];
+            string Folder2 = arguments[1];
+
+            if (!Directory.Exists(Folder1))
+            {
+                Console.WriteLine($"The folder {Folder1} from which we move the files doesn't exist");
+                Console.WriteLine("Program execution did not complete successfully");
+                return;
+            }
+
+            if (!Directory.Exists(Folder2))
+            {
+                Console.WriteLine($"The folder {Folder2} in which we move the filed doesn't exist");
+                Console.WriteLine("Program execution did not complete successfully");
+                return;
+            }
+
+            Console.WriteLine("Get the list of file formats to copy");
+
+            string[] Formats = new string[arguments.Length - 2];
+            Array.Copy(arguments, 2, Formats, 0, arguments.Length - 2);
+
+            Console.WriteLine("Get the list of files that match the file formats");
+
+            int filesMoved = 0;
+            foreach (string fileFormat in Formats)
+            {
+                Console.WriteLine($"Move files to the selected folder {Folder2}");
+                foreach (string file in Directory.EnumerateFiles(Folder1, fileFormat))
```

Comment:
take a look at the 3rd parameter: SearchOptions. it allows searching in subdirectories.

Reply:
Added SearchOption.AllDirectories, now moves files from subdirectories too.
foreach (string file in Directory.EnumerateFiles(Folder1, fileFormat, SearchOption.AllDirectories))